crypto/x509/x509_lu.c

```diff
@@ -321,9 +321,7 @@ int X509_STORE_CTX_get_by_subject(X509_STORE_CTX *vs, X509_LOOKUP_TYPE type,
     ret->type = tmp->type;
     ret->data.ptr = tmp->data.ptr;
 
-    X509_OBJECT_up_ref_count(ret);
-
-    return 1;
+    return X509_OBJECT_up_ref_count(ret);
```

Comment:
I think I need to set ret->type = X509_LU_NONE, for safety reasons,
because in X509_STORE_CTX_get_obj_by_subject and X509_STORE_CTX_get1_certs
we have code like
```C
    if (!X509_STORE_CTX_get_by_subject(vs, type, name, ret)) {
        X509_OBJECT_free(ret);
        return NULL;
    }
```

Reply:
BTW: the locking in this function is odd.  usually I increment references on objects while I hold a lock,
not after I release the lock.